### PR TITLE
SONAR-6993 Consume file duplication index to feed duplication block measures

### PR DESCRIPTION
**Pull request by julienlancelot**

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/computation/step/LoadCrossProjectDuplicationsRepositoryStep.java [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+    public CpdTextBlockToBlock(String fileKey) {
+      this.fileKey = fileKey;
+    }
+
+    @Override
+    public Block apply(@Nonnull CpdTextBlock duplicationBlock) {
+      return Block.builder()
+        .setResourceId(fileKey)
+        .setBlockHash(new ByteArray(duplicationBlock.getHash()))
+        .setIndexInFile(indexInFile++)
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/computation/step/LoadCrossProjectDuplicationsRepositoryStep.java]: MAJOR Extract this increment or decrement operator into a dedicated statement. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS881)

File: server/sonar-server/src/main/java/org/sonar/server/computation/step/LoadCrossProjectDuplicationsRepositoryStep.java [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+import org.sonar.api.utils.log.Loggers;
+import org.sonar.batch.protocol.output.BatchReport.CpdTextBlock;
+import org.sonar.db.DbClient;
+import org.sonar.db.DbSession;
+import org.sonar.db.duplication.DuplicationUnitDto;
+import org.sonar.duplications.block.Block;
+import org.sonar.duplications.block.ByteArray;
+import org.sonar.server.computation.analysis.AnalysisMetadataHolder;
+import org.sonar.server.computation.batch.BatchReportReader;
+import org.sonar.server.computation.component.*;
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/computation/step/LoadCrossProjectDuplicationsRepositoryStep.java]: MAJOR Explicitly import the specific classes needed. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2208)

File: server/sonar-server/src/main/java/org/sonar/server/computation/step/PersistCrossProjectDuplicationIndexStep.java [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    private void visitComponent(Component component) {
+      int indexInFile = 0;
+      try (CloseableIterator<BatchReport.CpdTextBlock> blocks = reportReader.readCpdTextBlocks(component.getReportAttributes().getRef())) {
+        while (blocks.hasNext()) {
+          BatchReport.CpdTextBlock block = blocks.next();
+          dbClient.duplicationDao().insert(session, new DuplicationUnitDto()
+            .setHash(block.getHash())
+            .setStartLine(block.getStartLine())
+            .setEndLine(block.getEndLine())
+            .setIndexInFile(indexInFile++)
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/computation/step/PersistCrossProjectDuplicationIndexStep.java]: MAJOR Extract this increment or decrement operator into a dedicated statement. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS881)

**Comment by SonarTech** [comment]
SonarQube analysis reported 5 issues:
* MAJOR 5 major

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MAJOR [Remove this unused "mybatis" private field.](https://github.com/SonarSource/sonarqube/blob/af6b44f07741ff605ca8fb8e07f32fd905358dbd/sonar-db/src/main/java/org/sonar/db/duplication/DuplicationDao.java#L34) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1068)
* MAJOR Add a 'package-info.java' file to document the 'snapshot' package (org.sonarsource.sonarqube:sonar-server:src/main/java/org/sonar/server/computation/snapshot) [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)

**Comment by sns-seb** [comment]
@julienlancelot PR back to you

**Comment by julienlancelot** [comment]
Feedback applied in branch feature/ce_cpd_feedbacks

**Pull request closed without merging**